src/dune_rules/enabled_if.ml

```diff
@@ -18,6 +18,7 @@ let common_vars_list =
   ; "profile"
   ; "ocaml_version"
   ; "context_name"
+  ; "arch_sixtyfour"
```

Comment:
I'm not sure how versioning is supposed to work here. For example, `(rule)` uses `~allowed_vars:Any` so this means that it will accept `arch_sixtyfour` even with lang dune 1.4 (this affects `context_name` too). @rgrinberg do you know how this is supposed to work?

Reply:
that's not correct. it means a dune file can be valid or not depending on the version of the dune binary, which we try to avoid.